Userland/Libraries/LibPDF/Filter.cpp

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright (c) 2021, Matthew Olsson <mattco@serenityos.org>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/Hex.h>
+#include <LibCompress/Deflate.h>
+#include <LibPDF/Filter.h>
+
+namespace PDF {
+
+Optional<ByteBuffer> Filter::decode(const ReadonlyBytes& bytes, const FlyString& encoding_type)
+{
+    if (encoding_type == "ASCIIHexDecode")
```

Comment:
```suggestion
    if (encoding_type == "ASCIIHexDecode"sv)
```

To avoid extra `strlen`'s, or you could preallocate these as FlyStrings in a common namespace 
to reduce the whole thing to a pointer compare.
(Applies to the ones below as well)

Reply:
Yeah I thought about mimicing LibJS's `CommonPropertyNamed.h`, might as well do that in this PR